pkg/gardenlet/controller/extensions/worker_sync.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	v1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (e *ExtensionStateSyncController) syncWorkerStateRequest(req reconcile.Request) (reconcile.Result, error) {
```

Comment:
I see that most of this code will be the same for all resources in the extensions API. Can't we implement this function more generically?